resource/modules/resource_match.cpp

```diff
@@ -644,25 +654,18 @@ static void match_request_cb (flux_t *h, flux_msg_handler_t *w,
     stringstream R;
     resource_ctx_t *ctx = getctx ((flux_t *)arg);
 
-    if (flux_msg_get_userid (msg, &userid) < 0)
-        goto error;
-
-    flux_log (h, LOG_INFO, "match requested by user (%u).", userid);
-
     if (flux_request_unpack (msg, NULL, "{s:s s:I s:s}", "cmd", &cmd,
                              "jobid", &jobid, "jobspec", &js_str) < 0)
         goto error;
-
     if (is_existent_jobid (ctx, jobid)) {
-        flux_log (h, LOG_INFO, "invalid jobid (%ld).", (intmax_t)jobid);
         errno = EINVAL;
+        flux_log_error (h, "existent job (%jd).", (intmax_t)jobid);
```

Comment:
```suggestion
        flux_log_error (h, "nonexistent job (%jd).", (intmax_t)jobid);
```

Further down in `info_request_cb`, the same nonexistent job error results in an `ENOENT`. I wonder if that fits better here too.

Reply:
Ah. Thanks. Yes, `ENOENT` should be used instead unless it has a side effect (I don't believe so). I will follow up.

Reply:
Looking at this once again, this is actually intended. In this case, the code is checking whether the job id already exists and if so it refuses the match request. So `EINVAL` should correct as well. So I will just push one commit that fixes the `done` label issue.

I was confused myself as well. So thank you for brining this up!